plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/WatchManager.java

```diff
@@ -271,9 +331,9 @@ public void connectionRemoved(IConnection connection) {
 			Connection conn = (Connection)connection;
 			synchronized (watches) {
 				watches.keySet().stream()
-					.filter(p->conn.ownsResource(p))
+					.filter(k->k.connection.equals(conn))
```

Comment:
@fbricon connection should really not consider token during evaluation.  I thought @adietish made changes a while back to address that.  I thought equality was determined to me server + username